src/range/Ranges.js

```diff
@@ -204,6 +214,109 @@ class Ranges extends Array {
     getAnnotations(options) {
         return GUI.annotations1D(this, options);
     }
+    /**
+     * Return an array of deltas and multiplicity for an index database
+     * @returns {Array} [{delta, multiplicity},...]
+     */
+    toIndex() {
+        var index = [];
+        this.compactPatterns();
+        for(var range of this) {
+            if (range.signal == undefined || range.signal == []) {
```

Comment:
It's better to be explicit. BTW it's impossible for `range.signal == []` to be true (`[]` creates a new array object. it cannot be equal to an existing value)